tests/sys/psa_crypto_cbor_encoder/test_psa_crypto_enc_dec_keypair.c

```diff
@@ -0,0 +1,113 @@
+#include "psa/crypto.h"
+#include "psa_crypto_cbor_encoder.h"
+#include "tests_psa_cbor_enc_dec.h"
+
+static uint8_t cbor_encoded_data[] = {
+    0x9f, 0x85, 0x00, 0x19, 0x71, 0x12, 0x19, 0x01,
+    0x00, 0x01, 0x82, 0x19, 0x30, 0x00, 0x1a, 0x06,
+    0x00, 0x06, 0x09, 0x82, 0x82, 0x18, 0x20, 0x98,
+    0x20, 0x18, 0x27, 0x18, 0xfc, 0x18, 0x4d, 0x18,
+    0x41, 0x18, 0xf3, 0x18, 0xde, 0x18, 0x49, 0x18,
+    0xf7, 0x18, 0x86, 0x18, 0x31, 0x18, 0x4b, 0x18,
+    0x74, 0x18, 0xaa, 0x18, 0x67, 0x18, 0xde, 0x18,
+    0x4b, 0x18, 0xa9, 0x18, 0x61, 0x18, 0xc3, 0x18,
+    0x8f, 0x18, 0x4f, 0x18, 0x89, 0x18, 0x6f, 0x04,
+    0x18, 0x5a, 0x18, 0x53, 0x18, 0x71, 0x02, 0x18,
+    0xb4, 0x18, 0x3d, 0x18, 0x20, 0x18, 0x39, 0x82,
+    0x18, 0x41, 0x98, 0x41, 0x04, 0x18, 0x56, 0x18,
+    0xcb, 0x18, 0x81, 0x18, 0xd1, 0x18, 0xcb, 0x18,
+    0xde, 0x18, 0x44, 0x18, 0xf5, 0x18, 0x1d, 0x18,
+    0xcc, 0x18, 0xb1, 0x18, 0x26, 0x02, 0x18, 0x67,
+    0x0d, 0x18, 0x76, 0x18, 0xdd, 0x18, 0xe7, 0x18,
+    0x84, 0x18, 0xed, 0x18, 0x8d, 0x18, 0x30, 0x18,
+    0x72, 0x18, 0x1c, 0x18, 0xca, 0x18, 0x50, 0x18,
+    0x59, 0x18, 0xf9, 0x18, 0x20, 0x18, 0xad, 0x18,
+    0x62, 0x18, 0x87, 0x18, 0x74, 0x18, 0x9e, 0x18,
+    0xc9, 0x18, 0xcb, 0x18, 0x26, 0x18, 0x75, 0x18,
+    0xc5, 0x18, 0x1b, 0x18, 0x69, 0x18, 0xa6, 0x18,
+    0x89, 0x18, 0x56, 0x10, 0x18, 0x2e, 0x18, 0x8f,
+    0x18, 0x6f, 0x18, 0x72, 0x18, 0x57, 0x18, 0xb9,
+    0x18, 0xb9, 0x18, 0x93, 0x18, 0xed, 0x18, 0x88,
+    0x18, 0x99, 0x18, 0xea, 0x18, 0xfd, 0x18, 0x53,
+    0x18, 0x82, 0x18, 0x3d, 0x18, 0xca, 0x18, 0xb6,
+    0x18, 0x41, 0xff
+};
+
+static size_t encoded_size;
+static psa_key_attributes_t attr;
+static psa_key_pair_slot_t slot;
+static psa_key_pair_slot_t dec_slot;
+
+static uint8_t cbor_enc[sizeof(cbor_encoded_data)];
+
+/**
+ * @brief   CBOR encoding of key pair slot should equal @c cbor_encoded_data
+ */
+static void test_encode_asymmetric_keypair_slot(void)
+{
+    psa_key_lifetime_t lifetime = 1;
+    psa_key_usage_t usage = PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH;
+
+    psa_set_key_lifetime(&attr, lifetime);
+    psa_set_key_algorithm(&attr, ECC_ALG);
+    psa_set_key_usage_flags(&attr, usage);
+    psa_set_key_type(&attr, ECC_KEY_TYPE);
+    psa_set_key_bits(&attr, ECC_KEY_SIZE);
+
+    slot.attr = attr;
+    memcpy(slot.key.privkey_data, privkey, sizeof(privkey));
+    slot.key.privkey_data_len = sizeof(privkey);
+
+    memcpy(slot.key.pubkey_data, pubkey, sizeof(pubkey));
+    slot.key.pubkey_data_len = sizeof(pubkey);
+
+    TEST_ASSERT_PSA(psa_encode_key_slot((psa_key_slot_t *)&slot, cbor_enc, sizeof(cbor_enc), &encoded_size));
+    TEST_ASSERT_EQUAL_INT(sizeof(cbor_encoded_data), encoded_size);
+    TEST_ASSERT_MESSAGE(1 == compare(cbor_enc, cbor_encoded_data,
+                                     encoded_size), "wrong cbor encoding");
+}
+
+/**
+ * @brief   Decoded key slot should equal key slot structure initialized
+ *          in @c test_encode_asymmetric_keypair_slot.
+ */
```

Comment:
Not sure if there is any guarantee on the order of the tests in embUnit. I think it would be better to refactor the slot structure initialization in a helper function which is used by both tests. Has the extra benefit of getting rid of the global variables.

The same applies to the other files.